### PR TITLE
feat(Page): securityDetails

### PR DESCRIPTION
**Pull request by trnkatomas**
Added feature securityDetails, the response in puppeteer now contains info about security state returned by Chrome DevTools Protocol. Slightly adjusted tests to be able to test this feature.

### CODE REVIEW

**Comment by trnkatomas** [comment]
I signed it!

**Comment by trnkatomas** [comment]
I signed it!

**Comment by Michaelknigth** [comment]
Tal vez deva eliminar elgunos archivos duplicados, como los contactos
esparcidos en  la maps app

File: docs/api.md [hunk 1]
```diff
@@ -2173,6 +2174,9 @@ Contains a boolean stating whether the response was successful (status in the ra
 #### response.request()
 - returns: <[Request]> A matching [Request] object.
 
+#### response.securityDetails()
+- returns: <[Object]> An object with security details associated with the response. All item are preserved in form as returned by Chrome DevTools protocol.
```
Comment (aslushnikov) [inline docs/api.md]: > All item are preserved in form as returned by Chrome DevTools protocol.

@trnkatomas there are fields that don't make sense to expose in the pptr api, e.g. `certificateId` and array of `Network.SignedCertificateTimestamp` objects.

What are your usecases for this?  Let's expose only the information you care about.

Comment (trnkatomas) [reply docs/api.md]: Hi, thanks for the comment, I actually need issuer, validity from and to and the subject of the certificate. Shall I just extract these information? I need the issuer for the information how much to trust the certificate and the time of validity for quite the same reason.

Comment (aslushnikov) [reply docs/api.md]: >  Shall I just extract these information?

@trnkatomas yes, let's extract it

> I need the issuer for the information how much to trust the certificate and the time of validity for quite the same reason.

Thank you for sharing your usecase.

Comment (aslushnikov) [inline docs/api.md]: can you please fix the doc to list explicitly what's in the securityDetails?

File: test/test.js [hunk 2]
```diff
@@ -1205,6 +1212,15 @@ describe('Page', function() {
...
       expect(response.url()).toContain('self-request.html');
     });
+    // TODO edit to vefify that results are retrieved
+    it('should support ignoreHTTPSErrors option', async({page_https, httpsServer}) => {
+      let error = null;
+      const response = await page_https.goto(httpsServer.EMPTY_PAGE).catch(e => error = e);
+      expect(error).toBe(null);
+      expect(response.ok()).toBe(true);
+      expect(response.securityDetails()).toBeTruthy();
+      expect(response.securityDetails().protocol).toBe('TLS 1.2');
```
Comment (aslushnikov) [inline test/test.js]: let's make SecurityDetails class and convert all fields to methods.

We don't have any getters/properties anywhere in the API.

File: test/test.js [hunk 3]
```diff
@@ -1205,6 +1212,15 @@ describe('Page', function() {
       expect(response.status()).toBe(200);
       expect(response.url()).toContain('self-request.html');
     });
+    // TODO edit to vefify that results are retrieved
+    it('should support ignoreHTTPSErrors option', async({page_https, httpsServer}) => {
```
Comment (aslushnikov) [inline test/test.js]: There are other ignoreHTTPErrors tests; let's move this one next to those. This way you'll avoid spawning second browser for all page tests.

Comment (trnkatomas) [reply test/test.js]: The only other test with `ignoreHTTPSErrors` I can spot is on line 111 testing the whole functionality of the response. Did you mean this occurrence?

**Comment by trnkatomas** [comment]
The appveyor seems to be failing because of some timeouts, is there any possibility to re-run the test?

**Comment by trnkatomas** [comment]
Please can someone take another look @aslushnikov? I did my best to fix the requested changes. Thanks

**Review by aslushnikov — APPROVED** [review]
@trnkatomas thank you for the work.

**Pull request merged**